galleries/examples/text_labels_and_annotations/line_with_text.py

```diff
@@ -28,9 +28,10 @@ def set_figure(self, figure):
         self.text.set_figure(figure)
         super().set_figure(figure)
 
-    def set_axes(self, axes):
-        self.text.set_axes(axes)
-        super().set_axes(axes)
+    @lines.Line2D.axes.setter
+    def axes(self, new_axes):
+        self.text.axes = new_axes
+        lines.Line2D.axes.fset(self, new_axes)
```

Comment:
This is an unusual enough pattern, that it probably warrants a comment explaining it. (Just a simple `# Override the axes property setter` and potentially `# Call the superclass property setter`)

It _works_, and is probably about the best you can do to override the property setter/call the super-class version of the property setter.

Some things I thought about which would have made some sense to me, but don't seem to work in this context:

- using `@MyLine.axes.setter` (`MyLine` is not available at class instantiation time)
   - I worried a bit about the setter accidentally overriding the parent classes setter, but that does _not_ happen)
- using `super().axes.fset(new_axes)` (`super().axes` resoves to the return of the getter, not the property descriptor)